torchgeo/datasets/agb_live_woody_density.py

```diff
@@ -112,15 +111,19 @@ def _verify(self) -> None:
 
     def _download(self) -> None:
         """Download the dataset."""
-        download_url(self.url, self.root, self.base_filename)
+        download_url(self.url, self.paths, self.base_filename)
 
-        with open(os.path.join(self.root, self.base_filename)) as f:
+        base_file = self.list_files(filename_glob=self.base_filename)[0]
+        with open(base_file) as f:
             content = json.load(f)
 
+        # TODO: This changes the behaviour from
+        #  unpacking in root to same dir as archive
```

Comment:
It's not clear to me exactly how all of this works. If `self.paths` is a list of files, how will `download_url` work? Do we need to add an assert statement somewhere that force paths to be a single dir? Or should we just use os.path.basename if it's a file?

Reply:
Yes, this is not intuitive. It works only because we know that the code instantiates this class with a str representing root. Why mypy does not complain that the type may not be as expected is unknown to me, any idea? 

This should annoy mypy:
```python
download_url(self.url, self.paths, self.base_filename)
```

I haven't been able come up with an elegant way of using `self.paths` as a root.